arches/app/frontend/webpack/webpack.common.js

```diff
@@ -0,0 +1,213 @@
+                ...acc, 
+                [subPathWithoutFileExtension]: { 'import': `${path}/${name}`, 'filename': `js/[name].js` } 
+            };
+        }
+    }, outerAcc);
+};
+
+const entryPointConfiguration = buildJavascriptFilepathLookup('../media/js', {});
+
+var javascriptRelativeFilepathToAbsoluteFilepathLookup = Object.keys(entryPointConfiguration).reduce((acc, path) => {
+    acc[path + '$'] = Path.resolve(__dirname, `../../media/js/${path}`);
+    return acc;
+}, {});
+
+const buildHtmlWebpackPluginList = function(path, outerAcc) {
+    return fs.readdirSync(path).reduce((acc, name) => {
+        if (fs.lstatSync(path + '/' + name).isDirectory() ) {
+            return buildHtmlWebpackPluginList(path + '/' + name, acc);
+        }
+        else {
+            const subPath = (path + '/' + name).slice(13);
+            const subPathWithoutFileExtension = subPath.slice(0, subPath.length - 4);
+
+            acc.push(new HtmlWebpackPlugin({
+                template: `../templates/${subPath}`, // relative path to the HTML files
+                filename: `templates/${subPath}`, // output HTML files
+                chunks:  [ Path.resolve(__dirname, `../../media/build/js/${subPathWithoutFileExtension}.js`) ] // respective JS files
+            }));
+            return acc;
+        }
+    }, outerAcc);
+};
+
+module.exports = {
+    entry: entryPointConfiguration,
+    output: {
+        path: Path.resolve(__dirname, '../../media/build'),
+        publicPath: '/foo/',
+        libraryTarget: 'amd-require',
+        clean: true,
+    },
+    optimization: {
+        // concatenateModules: true,
+        // flagIncludedChunks: true,
+        // removeAvailableModules: true,
+        // splitChunks: {  //  CURRENTLY DEFAULT VALUES
+        //     chunks: 'async',
+        //     minSize: 20000,
+        //     minRemainingSize: 0,
+        //     minChunks: 1,
+        //     maxAsyncRequests: 30,
+        //     maxInitialRequests: 30,
+        //     enforceSizeThreshold: 50000,
+        //     cacheGroups: {
+        //       defaultVendors: {
+        //         test: /[\\/]node_modules[\\/]/,
+        //         priority: -10,
+        //         reuseExistingChunk: true,
+        //       },
+        //       default: {
+        //         minChunks: 2,
+        //         priority: -20,
+        //         reuseExistingChunk: true,
+        //       },
+        //     },
+        //   },
+    },
+    plugins: [
+        new CleanWebpackPlugin(),
+        new CopyWebpackPlugin({ patterns: [{from: Path.resolve(__dirname, '../../media/img'), to: 'img'} ] }),
+        new webpack.ProvidePlugin({
+            jQuery:  Path.resolve(__dirname, '../../media/node_modules/jquery/dist/jquery.min'),
+            $:  Path.resolve(__dirname, '../../media/node_modules/jquery/dist/jquery.min'),
+        }),
+        new BundleTracker({ filename: './webpack-stats.json' }),
+    ].concat(buildHtmlWebpackPluginList('../templates', [])),
```

Comment:
`concat(buildHtmlWebpackPluginList('../templates', []))` results in creating an `htmlWebpackPlugin` for every HTML file in `templates`